packages/amplify-category-geo/src/service-walkthroughs/mapWalkthrough.ts

```diff
@@ -49,22 +49,30 @@ export const createMapWalkthrough = async (
 };
 
 export const mapNameWalkthrough = async (context: any): Promise<Partial<MapParameters>> => {
-    const mapNamePrompt = {
-        type: 'input',
-        name: 'name',
-        message: 'Provide a name for the Map:',
-        validate: context.amplify.inputValidation({
-            operator: 'regex',
-            value: '^[a-zA-Z0-9]+$',
-            onErrorMsg: 'You can use the following characters: a-z A-Z 0-9',
-            required: true,
-        }),
-        default: () => {
-            const [shortId] = uuid().split('-');
-            return `map${shortId}`;
-        },
-    };
-    return { name: (await inquirer.prompt([mapNamePrompt])).name as string };
+    let mapName;
+    while(!mapName) {
+        const mapNamePrompt = {
+            type: 'input',
+            name: 'name',
+            message: 'Provide a name for the Map:',
+            validate: context.amplify.inputValidation({
+                operator: 'regex',
+                value: '^[a-zA-Z0-9]+$',
+                onErrorMsg: 'You can use the following characters: a-z A-Z 0-9',
+                required: true,
+            }),
+            default: () => {
+                const [shortId] = uuid().split('-');
+                return `map${shortId}`;
+            },
+        };
+        const mapNameInput = (await inquirer.prompt([mapNamePrompt])).name as string;
```

Comment:
if you wanted to be the first consumer of the new `amplify-prompts` all of this could simplify to:
```js
const mapNameInput = await prompter.input('Provider a name for the Map', {validate: alphanumeric(), initial: uuid().split('-')})
```

Same also goes for places where you use `context.print.info()`. That could be changed to `printer.info()` from `amplify-prompts`

Reply:
There are many other places where we need to make the change. I can do a separate PR with migration to `amplify-prompts`. Thanks for contributing that package. Much needed!